### PR TITLE
feat(lanelet2_extension): format v2 bicycle lane doc

### PR DESCRIPTION
**Pull request by soblin**
## Description

depends https://github.com/autowarefoundation/autoware_lanelet2_extension/pull/21

## Related links

https://github.com/autowarefoundation/autoware_lanelet2_extension/pull/21

## Tests performed

document update

## Interface changes

none

## Effects on system behavior

none

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [X] I've confirmed the [contribution guidelines].
- [X] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [X] The PR follows the [pull request guidelines].
- [X] The PR has been properly tested.
- [X] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [X] There are no open discussions or they are tracked via tickets.
- [X] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Review by youtalk — APPROVED** [review]
LGTM

File: autoware_lanelet2_extension/docs/extra_lanelet_subtypes.md [hunk 1]
```diff
@@ -43,3 +43,29 @@ Sample pedestrian lane in .osm format is shown below:
...
+
+Although `bicycle_lane` subtype is already defined in the original [Lanelet2 documentation](https://github.com/fzi-forschungszentrum-informatik/Lanelet2/blob/master/lanelet2_core/doc/LaneletAndAreaTagging.md#subtype-and-location), Autoware compatible with `format_version>=2` supports `bicycle_lane` at component level.
+
+- refers: `road` lanelet adjacent to the objective `bicycle_lane` should share the same boundary LineString
+- specification:
+  - A `bicycle_lane` is connected to/from only `bicycle_lane` lanelets and is adjacent to a `road` lanelet. Namely in Left(Right) hand side traffic rules, it is on the left(right) side of `road` lanelet.
+- Although pullover on `bicycle_lane` is normally prohibited by the law, it is allowed to place a `road_shoulder` adjacent to a `bicycle_lane`. However the autoware implementation may not fully support appropriate behavior in such cases.
+- The treatment of _bicycle guide painting_, or the shift-arrow shaped painting on the side of the road without explicit separating line, is open to discussion. We can either identity them as part of the vehicle road or separate them as distinct bicycle lanes.
+
+![Bicycle lane](bicycle_lane.svg)
```
Comment (mitsudome-r) [inline autoware_lanelet2_extension/docs/extra_lanelet_subtypes.md:57]:
```suggestion
![Bicycle lane](bicycle_lane.drawio.svg)
```

File: autoware_lanelet2_extension/docs/extra_lanelet_subtypes.md [hunk 2]
```diff
@@ -43,3 +43,29 @@ Sample pedestrian lane in .osm format is shown below:
...
   </relation>
 ```
+
+## Bicycle Lane (format_version >= 2)
+
+Although `bicycle_lane` subtype is already defined in the original [Lanelet2 documentation](https://github.com/fzi-forschungszentrum-informatik/Lanelet2/blob/master/lanelet2_core/doc/LaneletAndAreaTagging.md#subtype-and-location), Autoware compatible with `format_version>=2` supports `bicycle_lane` at component level.
+
+- refers: `road` lanelet adjacent to the objective `bicycle_lane` should share the same boundary LineString
+- specification:
+  - A `bicycle_lane` is connected to/from only `bicycle_lane` lanelets and is adjacent to a `road` lanelet. Namely in Left(Right) hand side traffic rules, it is on the left(right) side of `road` lanelet.
```
Comment (mitsudome-r) [inline autoware_lanelet2_extension/docs/extra_lanelet_subtypes.md:53]: Shouldn't the first item go under specification? What does `refers` mean in this context?

File: autoware_lanelet2_extension/docs/extra_lanelet_subtypes.md [hunk 3]
```diff
@@ -43,3 +43,29 @@ Sample pedestrian lane in .osm format is shown below:
...
+- Although pullover on `bicycle_lane` is normally prohibited by the law, it is allowed to place a `road_shoulder` adjacent to a `bicycle_lane`. However the autoware implementation may not fully support appropriate behavior in such cases.
```
Comment (mitsudome-r) [inline autoware_lanelet2_extension/docs/extra_lanelet_subtypes.md:54]: It's slightly abrupt to start discussing about pullover behavior in this line.

My suggestion is to maybe explain the expected usecase of bicycle_lane after/before the first paragraph of Bicycle Lane section

e.g.,
In Autoware, bicycle lanes are used to identify paths dedicated to cyclists. These lanes can assist in predicting the future paths of bicycles and indicate lanes that Autoware vehicles should avoid or keep clear of.

File: autoware_lanelet2_extension/docs/extra_lanelet_subtypes.md (hunk 2 above)
Comment (soblin) [reply autoware_lanelet2_extension/docs/extra_lanelet_subtypes.md:53]: Other sections like "Road shoulder subtype" and "Pedestrian lane subtype" have the same "refer" description so I followed then. But yeah they are ambiguous.

**Review by mitsudome-r — APPROVED** [review]

**Pull request merged**